src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -2189,6 +2201,73 @@ public void setSensitivityAnalysisParametersValues(UUID studyUuid, SensitivityAn
         notificationService.emitElementUpdated(studyUuid, userId);
     }
 
+    public Integer fetchFiltersComplexity(UUID studyUuid, SensitivityAnalysisParametersInfos sensitivityAnalysisParametersValues, String userId) {
+        Map<Integer, Map<String, List<UUID>>> containerIdsMap = idsMapInitializer();
+        filterIdsBuilder(sensitivityAnalysisParametersValues, containerIdsMap);
+        UUID networkUuid = networkStoreService.getNetworkUuid(studyUuid);
+        return filterService.fetchFiltersComplexity(containerIdsMap, networkUuid);
+    }
+
+    private Map<Integer, Map<String, List<UUID>>> idsMapInitializer() {
+        Map<Integer, Map<String, List<UUID>>> containerIdsMap = new HashMap<>();
+
+        containerIdsMap.put(0, mapInit(new HashMap<>()));
+        containerIdsMap.put(1, mapInit(new HashMap<>()));
+        containerIdsMap.put(2, mapInit(new HashMap<>()));
+        containerIdsMap.put(3, mapInit(new HashMap<>()));
+        return containerIdsMap;
+    }
+
+    private Map<String, List<UUID>> mapInit(Map<String, List<UUID>> identifierIdsMap) {
+        identifierIdsMap.put(MONITORED_BRANCHS, new ArrayList<>());
+        identifierIdsMap.put(INJECTIONS, new ArrayList<>());
+        identifierIdsMap.put(CONTINGENCIES, new ArrayList<>());
+        return identifierIdsMap;
+    }
+
+    private static void filterIdsBuilder(SensitivityAnalysisParametersInfos sensitivityAnalysisParametersValues, Map<Integer, Map<String, List<UUID>>> containersIds) {
+        sensitivityAnalysisParametersValues.getSensitivityInjectionsSet().forEach(sensitivityInjectionsSet -> {
+            if (sensitivityInjectionsSet.isActivated()) {
+                sensitivityInjectionsSet.getMonitoredBranches().forEach(monBranch ->
+                        containersIds.get(0).get(MONITORED_BRANCHS).add(monBranch.getContainerId()));
+                sensitivityInjectionsSet.getInjections().forEach(monBranch ->
+                        containersIds.get(0).get(INJECTIONS).add(monBranch.getContainerId()));
+                sensitivityInjectionsSet.getContingencies().forEach(contingency ->
+                        containersIds.get(0).get(CONTINGENCIES).add(contingency.getContainerId()));
```

Comment:
You can create  one method to handle all equipments.
```suggestion
            private void getContainersIds(List<EquipmentsContainer> branches, List<EquipmentsContainer> injectors, List<EquipmentsContainer> contin, Map<Integer, Map<String, List<UUID>>> containersIds, int index){
        branches.forEach(monBranch ->
                containersIds.get(index).get(MONITORED_BRANCHS).add(monBranch.getContainerId()));
        injectors.forEach(monBranch ->
                containersIds.get(index).get(INJECTIONS).add(monBranch.getContainerId()));
        contin.forEach(contingency ->
                containersIds.get(index).get(CONTINGENCIES).add(contingency.getContainerId()));
    }
```